### PR TITLE
[zh-cn]: update the translation of WeakMap constructor

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Same as the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#16726

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/weakmap/weakmap/index.md [hunk 1]
```diff
@@ -1,58 +1,58 @@
...
 wm1.get(o2); // "azerty"
-wm2.get(o2); // undefined,wm2中没有o2这个键
-wm2.get(o3); // undefined,值就是undefined
+wm2.get(o2); // undefined，因为 wm2 上没有 o2 的键
+wm2.get(o3); // undefined，因为设置的值就是 undefined
 
 wm1.has(o2); // true
 wm2.has(o2); // false
-wm2.has(o3); // true (即使值是undefined)
+wm2.has(o3); // true（即使值本身是 ‘undefined’）
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/weakmap/weakmap/index.md:48]:
```suggestion
wm2.has(o3); // true（即使值是 undefined）
```

File: files/zh-cn/web/javascript/reference/global_objects/weakmap/weakmap/index.md [hunk 2]
```diff
@@ -1,58 +1,58 @@
...
+new WeakMap(iterable)
 ```
 
+> **备注：** `WeakMap()` 构造函数只能使用 [`new`](/zh-CN/docs/Web/JavaScript/Reference/Operators/new) 调用。不使用 `new` 而直接调用会抛出 {{jsxref("TypeError")}}。
+
 ## 参数
 
 - `iterable`
-  - Iterable 是一个数组（二元数组）或者其他可迭代的且其元素是键值对的对象。每个键值对会被加到新的 WeakMap 里。null 会被当做 undefined。
+  - : 一个 [`Array`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array) 或者其他实现了 [`@@iterator`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/@@iterator) 方法的可迭代对象，该方法返回一个迭代器对象，该对象会产生一个二元数组对象，其第一个元素是将被用作 `WeakMap` 键的对象，第二个元素是与该键相关联的值。每个键值对都会被添加到新的 `WeakMap` 对象中。`null` 被视为 `undefined`。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/weakmap/weakmap/index.md:22]:
```suggestion
  - : 一个 [`Array`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array) 或者其他实现了 [`@@iterator`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/@@iterator) 方法的可迭代对象，该方法返回一个迭代器对象，该对象会产生一个二元类数组对象，其第一个元素是将被用作 `WeakMap` 键的对象，第二个元素是与该键相关联的值。每个键值对都会被添加到新的 `WeakMap` 对象中。`null` 会被视为 `undefined`。
```

File: files/zh-cn/web/javascript/reference/global_objects/weakmap/weakmap/index.md [hunk 3]
```diff
@@ -63,11 +63,11 @@ wm1.has(o1);   // false
...
 {{Compat}}
 
-## 相关链接
+## 参见
 
-- A polyfill of `WeakMap` is available in [`core-js`](https://github.com/zloirock/core-js#weakmap)
-- [`WeakMap` in the JavaScript guide](/zh-CN/docs/Web/JavaScript/Guide/Keyed_collections#weakmap对象)
-- [Hiding Implementation Details with ECMAScript 6 WeakMaps](https://fitzgeraldnick.com/weblog/53/)
+- [`core-js` 中 `WeakMap` 的 polyfill](https://github.com/zloirock/core-js#weakmap)
+- [JavaScript 指南中的 `WeakMap`](/zh-CN/docs/Web/JavaScript/Guide/Keyed_collections#weakmap_object)
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/weakmap/weakmap/index.md:69]:
```suggestion
- [JavaScript 指南中的 `WeakMap`](/zh-CN/docs/Web/JavaScript/Guide/Keyed_collections#weakmap_对象)
```

**Review by yin1999 — APPROVED** [review]
Thanks @JasonLamv-t

**Pull request merged**